janis_bioinformatics/data_types/vcf.py

```diff
@@ -24,6 +34,63 @@ def doc(self):
     Documentation: https://samtools.github.io/hts-specs/VCFv4.3.pdf
     """.strip()
 
+    @classmethod
+    def md5_without_header(cls, file_path: str, headers_to_remove: List[str]) -> str:
+        """
+        Compute md5 of a vcf file with unwanted headers removed
+
+        :param file_path: path to the file
+        :type file_path: str
+        :param headers_to_remove: headers to be removed before computing md5
+        :type headers_to_remove: List[str]
+        :return: md5
+        :rtype: str
+        """
+        with open(file_path, "r") as f:
+            hash_md5 = hashlib.md5()
+            while True:
+                line = f.readline()
+                if not line:
+                    break
+                if all(("##" + header) not in line for header in headers_to_remove):
+                    hash_md5.update(line.encode())
+        return hash_md5.hexdigest()
+
+    @classmethod
+    def basic_test(
+        cls,
+        tag: str,
+        min_size: int,
+        line_count: int,
```

Comment:
We almost always need run the tool at least once to understand what outputs we can expect before we can do any meaningful test. And `min_size` and `line_count `were observed from multiple runs. With vcf files, at least in my local testing, their number of lines were always the same (although exact content could vary).

If we assume we have no prior knowledge of the outputs, should we also change how `min_size` is handled? Because with such assumption, we can only check if file size is bigger than 0.

Reply:
I think we definitely have to assume prior knowledge of the expected output file.
But, the reason why i think we might prefer to make `line_count` optional here is so that if people already provide md5, they don't necessarily have to provide line_count anymore. But, it's not a big issue, just a suggestion.

Reply:
I've corrected it and other same cases below.